link_shortener/commands/update.py

```diff
@@ -43,19 +44,28 @@ async def update_link(request, link_id, data):
 
         link_update = links.update().where(links.columns['id'] == link_id)
 
-        if data['password']:
-            salt = os.urandom(32)
-            password = hashlib.pbkdf2_hmac(
-                'sha256',
-                data['password'].encode('utf-8'),
-                salt,
-                100000
-            )
+        if data['password'] == config('DEFAULT_PASSWORD'):
+            await conn.execute(link_update.values(
+                endpoint=data['endpoint'] if data['endpoint'] else link_data['endpoint'],
+                url=data['url'] if data['url'] else link_data['url'],
+                switch_date=data['switch_date']
+            ))
+        else:
             if link_data.password:
-                await conn.execute(salts.update().where(
-                    salts.columns['link_id'] == link_id
-                ).values(salt=salt))
+                await conn.execute(salts.delete().where(
+                    salts.columns['link_id'] == link_data.id
+                ))
+
+            if not data['password']:
+                password = None
             else:
```

Comment:
Snad dobrý, pokud sem se do ifů sám nezamotal 🙂 Jen tu máš znovu if not, else. Tak prosím jen přehoď pořadí ať máš if, else